### DISCUSSION TITLE
7530 7.28 Flashing

### DISCUSSION THREAD

**Discussion by miathedev** [Freetz-NG/freetz-ng#392] [Allgemein]
Hi!

i tried to flash my FritzBox 7530 with firmare version 7.28 with the current freetz(-ng) out of the main branch.

Ive been told that the 7530 wont accept web flashing - ok.

Does someone got the flash process using ftp fully working and is able to give me his/her steps?
I tried a lot of ways, using the eva tools and push-firmware of course. I never ever succeeded. I think i made sth wrong from the beginning.

Please if possible someone give me the steps needed using the "Freetz-Linux" - i hope the VM is independent from host issues. (I tried my host too on my first attempt)

I would really appreciate any help!
With kind regards,
mia

**Comment by fda77** [comment]
While you tried it there where some error messages. Read and then fix them

**Comment by conn-fused** [comment]
I'm in the same situation. I can't flash a 7530 with freetz-ng despite working extensively with push-firmware from a linux machine. Here's a little more information:

The push-firmware command always fails.

My closest attempt used the command:

tools/push_firmware images/7530_07.29.FREETZ-NG.image -cmd ftp -mr -ram 0 -ip 192.168.178.1

With this command I was able to detect the box, but the command fails like this:

> No reply from box, assuming switch-off or restart. Trying to re-detect box.
Waiting ............. found!
Detected memsize: '0x10000000'
tools/push_firmware: line 92: 0 - ( 28547328 % 0 ) : division by 0 (error token is "0 ) ")

I had suggestion elsewhere to simply use: 

`tools/push_firmware images/7530_07.29.FREETZ-NG.image -cmd ftp`

This does not work. It exits with:

> "tools/push_firmware: line 92: 0 - ( 28547328 % 0 ) : division by 0 (error token is "0 ) ")"

Line 92 in the push_firmware script is: 
`"ALIBYTES="$(( $ALIBYTES - ( $FILESIZE % $ALIBYTES ) ))""`

It looks like ALIBYTES refers to the alignment, something that the script should use only in ram-boot or fit-boot modes. Because the alignment is set to zero this line produces division by zero. I have no idea what the alignment should be for a 7530 though ...
